server/utils/dateUtils.ts

```diff
@@ -271,6 +281,18 @@ export const uiDateOrDateEmptyMessage = (
   return 'No date supplied'
 }
 
+export const dateIsValid = (date: string) => {
```

Comment:
I'm amazed that we don't have this already!

Perhaps call this `dateIsValidIso` as it's not clear from the name, how it's validating.